common/changeset/storage_changeset.go

```diff
@@ -330,27 +327,24 @@ func (b StorageChangeSetBytes) Walk(f func(k, v []byte) error) error {
 
 	var addressHashID uint32
 	var id int
-	k := make([]byte, common.HashLength*2+common.IncarnationLength)
 	for i := 0; i < numOfUniqueElements; i++ {
 		var (
 			startKeys int
 			endKeys   int
 		)
 
 		if i > 0 {
-			startKeys = int(binary.BigEndian.Uint32(b[4+i*(common.HashLength)+(i-1)*4 : 4+i*(common.HashLength)+(i)*4]))
+			startKeys = int(binary.BigEndian.Uint32(b[4+i*(common.AddressLength)+(i-1)*4 : 4+i*(common.AddressLength)+(i)*4]))
 		}
-		endKeys = int(binary.BigEndian.Uint32(b[4+(i+1)*(common.HashLength)+i*4:]))
-		addrHash := b[4+i*(common.HashLength)+i*4:]
+		endKeys = int(binary.BigEndian.Uint32(b[4+(i+1)*(common.AddressLength)+i*4:]))
+		addrBytes := b[4+i*(common.AddressLength)+i*4:]
 		incarnation := DefaultIncarnation
 		if inc, ok := notDefaultIncarnations[addressHashID]; ok {
 			incarnation = inc
 		}
 
 		for j := startKeys; j < endKeys; j++ {
-			copy(k[:common.HashLength], addrHash[:common.HashLength])
-			copy(k[common.HashLength+common.IncarnationLength:2*common.HashLength+common.IncarnationLength], b[keysStart+j*common.HashLength:])
-			binary.BigEndian.PutUint64(k[common.HashLength:], ^incarnation)
+			k := dbutils.PlainGenerateCompositeStorageKey(common.BytesToAddress(addrBytes[:common.AddressLength]), incarnation, common.BytesToHash(b[keysStart+j*common.HashLength:keysStart+j*common.HashLength+common.HashLength]))
```

Comment:
The point of having `k` here was to reduce allocations

Reply:
yeah, I guessed that, but forgot to fix it back